homeworks/01-bubble-sort/src/App.tsx

```diff
@@ -1,26 +1,103 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import { BarSet } from './components/BarSet';
+
+import { createRandomNumberArray } from './utils/appUtils';
+
+const MAX_NUMBER = 100;
+const NUMBER_COUNT = 40;
+const SORT_TIMEOUT = 75;
+
+interface AppProps {}
+
+interface AppState {
+  sorting: boolean,
+  sortingState: {
+    iteration: number,
+    currentIndex: number,
+  },
+  numbers: number[],
+  solved: boolean,
+}
+
+class App extends React.Component<{}, AppState> {
+  constructor(props: AppProps) {
+    super(props);
+    this.state = {
+      sorting: false,
+      sortingState: { iteration: 0, currentIndex: 0 },
+      numbers: createRandomNumberArray(MAX_NUMBER, NUMBER_COUNT),
+      solved: false,
+    };
+  }
```

Comment:
You can write just the following:
```tsx
    state = {
      sorting: false,
      sortingState: { iteration: 0, currentIndex: 0 },
      numbers: createRandomNumberArray(MAX_NUMBER, NUMBER_COUNT),
      solved: false,
    };
```

Reply:
I get an error, if not using constructor:
> State initialization should be in a constructor  react/state-in-constructor

Reply:
disabled this rule, initialised state in the component beginning, remove useless AppProps interface